sympy/assumptions/ask.py

```diff
@@ -463,22 +437,17 @@ def ask_full_inference(proposition, assumptions, known_facts_cnf):
 def register_handler(key, handler):
     """
     Register a handler in the ask system. key must be a string and handler a
-    class inheriting from AskHandler::
-
-        >>> from sympy.assumptions import register_handler, ask, Q
-        >>> from sympy.assumptions.handlers import AskHandler
-        >>> class MersenneHandler(AskHandler):
-        ...     # Mersenne numbers are in the form 2**n - 1, n integer
-        ...     @staticmethod
-        ...     def Integer(expr, assumptions):
-        ...         from sympy import log
-        ...         return ask(Q.integer(log(expr + 1, 2)))
-        >>> register_handler('mersenne', MersenneHandler)
-        >>> ask(Q.mersenne(7))
-        True
+    class inheriting from AskHandler.
+
+    .. note:: Deprecated in Sympy 1.8.
 
     """
-    # Will be deprecated
+    SymPyDeprecationWarning(
+        feature="register_handler() function",
+        useinstead="multipledispatch handler of Predicate",
+        issue=20656,
```

Comment:
The issue number should refer to an issue that is for the future removal of the deprecated feature. That issue should be written specifically to explain:

1. What is deprecated
2. What the new version is
3. How to change code that uses the deprecated API
4. When the deprecated codepaths will be removed completely.

The issue should have the "deprecation removal" label.

The wording of the linked issue should be written to make sense to a user who sees the warning. Bear in mind that the user might not have any idea which part of their code (or any other code they are using) has triggered the warning so there needs to be a clear explanation.